### ISSUE TITLE
Migrate the modules to use vanity url go.opentelemetry.io/collector-contrib

### ISSUE THREAD

**Issue by dmitryax** [open-telemetry/opentelemetry-collector-contrib#21469]
### Is your feature request related to a problem? Please describe.

Similar to how it's done in the core repo, we should use vanity URLs for contrib modules:

github.com/open-telemetry/opentelemetry-collector-contrib -> go.opentelemetry.io/collector-contrib

Brought by @Aneurysm9 in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/21272#discussion_r1181930807

> Regarding the go.opentelemetry.io metadata proxy, it takes data from [here](https://github.com/open-telemetry/opentelemetry-go-vanityurls/blob/main/go.opentelemetry.io/vanity.yaml) and adding an entry for collector-contrib should be easy. I believe there are issues with hosting and management that are being worked through ATM, but I would expect the more significant impediment to its use by collector contrib to simply be the accumulated mass of code already built working with modules at the existing import paths that would need to change.

### Describe the solution you'd like

-

**Comment by Aneurysm9** [comment]
@damemi had mentioned that the existing vanity URL redirect implementation is hosted by Google and needs to be transferred to OTel GC/TC or CNCS operation.  I do not know the status or timelines of that effort.

Beyond that, what would be the impact to users of these modules if we change the import path?  What would the migration path look like?

**Comment by damemi** [comment]
> @damemi had mentioned that the existing vanity URL redirect implementation is hosted by Google and needs to be transferred to OTel GC/TC or CNCS operation. I do not know the status or timelines of that effort.

Yeah, I wrote up some details about how @MrAlias hit this in https://github.com/open-telemetry/opentelemetry-go-vanityurls/pull/17, and the steps we took to finally publish it in [this doc proposal](https://docs.google.com/document/d/1-qijBPd-68Zqi3Xom6RtFtKFOD3jdyYr8NAcn4vAlJo/edit#heading=h.vli04prw9vfd).

Now that the service account in the project has been given the necessary IAM roles, you *should* be able to deploy a new URL without needing a Googler to manually do anything with the project. 

But thinking longer term, it shouldn't be just Googlers with access to that project. I think @mtwo was planning to bring this up at the GC meeting.

**Comment by mx-psi** [comment]
> Beyond that, what would be the impact to users of these modules if we change the import path? What would the migration path look like?

An important question to me here is if this would mean a change for all ocb manifests (I expect the answer to be "yes")

**Comment by codeboten** [comment]
@mx-psi this sounds like something we could make the builder capable of managing for users to help alleviate the pain of the transition period.

**Comment by dmitryax** [comment]
> Beyond that, what would be the impact to users of these modules if we change the import path? What would the migration path look like?

@Aneurysm9 I believe it's only a one-time problem for the users that want to update the otel collector contrib dependencies in their repos. All the dependencies on the existing versions will continue working.

We can mitigate the migration in the builder by renaming the modules under the hood, as @codeboten mentioned in https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21382#issuecomment-1536443409. For users that don't use the builder, maybe we can provide a shell command to replace import paths in their repos.

Please let us know what other complications you can think of.

**Comment by codeboten** [comment]
Discussed at the 16-Sept-2026 SIG meeting, if we wanted to move away from github at any point in time, having the vanity url would make this easier (never mind the rest of the pile of work required 😰). Making this change would break end users that don't use the builder (assuming we support aliasing in the builder).

Will wait to see if there is any support on this issue (either through votes or through comments) to tackle this work, if not, I would vote to remove the vanity url altogether.

**Comment by mx-psi** [comment]
I personally don't feel like this is worth the effort given the amount of migration work that we and others would have to do. 

The main benefit I see is marking some modules are more 'official'. I would maybe want to do that with some set of blessed components (high priority components?), but I feel like the benefit is relatively small.

I am not completely opposed to it if somebody else is willing to do the work, the only thing I feel strongly about is that OCB users should have a smooth transition.

**Comment by andrzej-stencel** [comment]
I'm in favor of doing this. Makes builder manifest cleaner and applies good practices in terms of managing module URLs. In principle, the module URLs should be owned by the project.

I recognize this is a breaking change for users (especially non-OCB), who need to update the import paths. On the other hand, all that's needed to fix this is a regex replace command, right? (Admittedly, possibly across various repositories and systems in big deployments).

Ideally, this should be possible as a gradual update, with a version (or more than one) of contrib modules working with either URL. It looks like this is not possible with current Go tooling?

**Comment by atoulme** [comment]
Can we do it for 1.0 modules?

**Comment by codeboten** [comment]
> Can we do it for 1.0 modules?

This is definitely an option, my 2 cents is that inconsistencies would make things even more confusing for end users, with some modules in contrib on a vanity url because they're 1.x, and modules in core being 0.x and still using a vanity url.
